Load the snakemake module before running in job script, with added environment variable SNAKEMAKE_LOAD_MODULE
Always use the python executable defined by the environment
Author: Samuel Moors, Vrije Universiteit Brussel (VUB)
diff -ur snakemake-5.7.1.orig/snakemake/executors.py snakemake-5.7.1/snakemake/executors.py
--- snakemake-5.7.1.orig/snakemake/executors.py	2019-10-16 12:04:28.000000000 +0200
+++ snakemake-5.7.1/snakemake/executors.py	2019-11-03 16:45:49.435118155 +0100
@@ -505,8 +505,9 @@
         if exec_job is None:
             self.exec_job = "\\\n".join(
                 (
+                    os.getenv('SNAKEMAKE_LOAD_MODULE', ':') + ' && ',
                     "cd {workflow.workdir_init} && " if assume_shared_fs else "",
-                    "{sys.executable} " if assume_shared_fs else "python ",
+                    "python ",
                     "-m snakemake {target} --snakefile {snakefile} ",
                     "--force -j{cores} --keep-target-files --keep-remote ",
                     "--wait-for-files {wait_for_files} --latency-wait {latency_wait} ",
